### ISSUE TITLE
RS0030 Symbol not reported: Task.Delay(System.Int32, System.Threading.CancellationToken)

### ISSUE THREAD

**Issue by Kuinox** [dotnet/roslyn-analyzers#4473]
### Analyzer

**Diagnostic ID**: [RS0030]
### Analyzer source

```xml
<PackageReference Include="Microsoft.CodeAnalysis.BannedApiAnalyzers" Version="3.3.1" PrivateAssets="All" />
```

### Describe the bug
With the following configuration: 
```
M:System.Threading.Tasks.Task.Delay(System.Int32); Use the time handling interfaces instead.
M:System.Threading.Tasks.Task.Delay(System.Int32, System.Threading.CancellationToken); Use the time handling interfaces instead.
```

I got this:  
```csharp
await Task.Delay(10, default); //no warn, still bugged with "new System.Threading.CancellationToken()" too.
await Task.Delay(10); //warn
```

### Steps To Reproduce

https://github.com/Kuinox/BannedApiAnalyzerRepro

### Expected behavior

Both Task.Delay calls should emit a warning.

### Actual behavior

Only Task.Delay(int) emit a warning.

**Comment by Evangelink** [comment]
Took me ages to figure out the issue...

Replace `M:System.Threading.Tasks.Task.Delay(System.Int32, System.Threading.CancellationToken); Use the time handling interfaces instead.` with `M:System.Threading.Tasks.Task.Delay(System.Int32,System.Threading.CancellationToken); Use the time handling interfaces instead.` (no space between the parameters) and you'll get your issues :)

@mavasani I think I should try to focus on implementing this validation analyzer that would report on invalid lines (in this case it would report that we could not find the symbol associated with the declaration on the second line).

I am still closing this ticket as fixed.

**Comment by Kuinox** [comment]
Trimming the input would be great, no?  
This is a config, and a stupid user like me will add space because I replicated code style that I use in my code.

**Comment by Evangelink** [comment]
@Kuinox We should definitely improve the user experience around user configuration but we need to think about some global solution. 

The specific trim you are talking about could be one of the solution but I'd like to get some validation from core team on the fact we can remove all whitespaces or if we need some more clever parsing.

Tagging @mavasani

**Comment by Kuinox** [comment]
Ok! 
Anyway, thanks for figuring out my issue, have a good day.

**Comment by Evangelink** [comment]
You too :)

**Comment by mavasani** [comment]
> I think I should try to focus on implementing this validation analyzer that would report on invalid lines (in this case it would report that we could not find the symbol associated with the declaration on the second line)

Yes, that would be a good addition.

It might be good to even open a Roslyn compiler feature request to enhance the following public APIs on `DocumentationCommentId` to ignore redundant whitespaces:
1. [DocumentationCommentId.GetSymbolsForDeclarationId](https://github.com/dotnet/roslyn/blob/d66c12164c0da3841f86b165872ea1b55942c575/src/Compilers/Core/Portable/DocumentationCommentId.cs#L82-L85)
2. [DocumentationCommentId.GetFirstSymbolForDeclarationId](https://github.com/dotnet/roslyn/blob/d66c12164c0da3841f86b165872ea1b55942c575/src/Compilers/Core/Portable/DocumentationCommentId.cs#L133-L136)
3. [DocumentationCommentId.GetSymbolsForReferenceId](https://github.com/dotnet/roslyn/blob/d66c12164c0da3841f86b165872ea1b55942c575/src/Compilers/Core/Portable/DocumentationCommentId.cs#L160-L163)
4. [DocumentationCommentId.GetFirstSymbolForReferenceId](https://github.com/dotnet/roslyn/blob/d66c12164c0da3841f86b165872ea1b55942c575/src/Compilers/Core/Portable/DocumentationCommentId.cs#L216-L219)
